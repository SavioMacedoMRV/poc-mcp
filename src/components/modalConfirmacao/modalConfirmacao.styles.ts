import styled from 'styled-components'
import {cores} from 'resources/cores'
import {media} from 'resources/media'
import {margens} from 'resources/margens'
import {hexToRgba} from 'helpers/cores'
import {Botao} from 'components/botao'
import {Icone} from 'components/icone'
import {Texto} from 'components/texto'
import CancelarSVG from 'icons/iCancelar.svg'

export const BotaoCancelar = styled(Botao).attrs({
  altura: 48,
  largura: 144,
  textoProps: {tamanho: 16},
  texto: 'Cancelar',
  corTexto: cores.neutralXXLight,
  cor: cores.neutralDark,
})`
  border-radius: ${margens.small}px;
`

export const BotaoConfirmar = styled(BotaoCancelar).attrs({
  texto: 'Confirmar',
  corTexto: cores.supportFeedbackError,
  cor: cores.background02,
})``

export const ContainerBotoes = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
`

export const IconeCancelar = styled(Icone).attrs({
  altura: 48,
  icone: CancelarSVG,
  cor: cores.neutralXDark,
})``

export const ModalContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  gap: ${margens.large}px;
  padding: ${margens.large}px;
  width: 350px;
  border-radius: ${margens.large}px;
  background-color: ${cores.neutralXXLight};

  @media ${media.tablet} {
    width: 300px;
  }
`

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1;
  width: 100%;
  height: 100%;
  background-color: ${hexToRgba(cores.neutralDark, 0.5)};
  display: flex;
  justify-content: center;
  align-items: center;
`

export const TextoMensagem = styled(Texto).attrs({
  tamanho: 14,
  cor: cores.neutralDark,
  alinhamento: 'center',
})``

export const TextoTitulo = styled(Texto).attrs({
  tamanho: 18,
  estilo: 'semibold',
  cor: cores.neutralXDark,
})``
