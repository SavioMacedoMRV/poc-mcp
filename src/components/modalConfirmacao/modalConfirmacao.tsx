import React from 'react'
import * as S from './modalConfirmacao.styles'

interface ModalProps {
  exibir: boolean
  titulo: string
  mensagem: string
  aoCancelar: () => void
  aoConfirmar: () => void
}

export const ModalConfirmacao = ({
  exibir,
  titulo,
  mensagem,
  aoCancelar,
  aoConfirmar,
}: Readonly<ModalProps>) => {
  return (
    exibir && (
      <S.ModalOverlay
        data-testid="modal-confirmacao-overlay"
        onClick={aoCancelar}
      >
        <S.ModalContent data-testid="modal-confirmacao-content">
          <S.IconeCancelar />
          <S.TextoTitulo>{titulo}</S.TextoTitulo>
          <S.TextoMensagem>{mensagem}</S.TextoMensagem>
          <S.ContainerBotoes>
            <S.BotaoConfirmar
              data-testid="btn-modal-confirmacao-confirmar"
              aoClicar={aoConfirmar}
            />
            <S.BotaoCancelar
              data-testid="btn-modal-confirmacao-cancelar"
              aoClicar={aoCancelar}
            />
          </S.ContainerBotoes>
        </S.ModalContent>
      </S.ModalOverlay>
    )
  )
}
