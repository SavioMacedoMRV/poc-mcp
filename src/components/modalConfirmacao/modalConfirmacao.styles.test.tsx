import {render} from '@testing-library/react'
import * as S from './modalConfirmacao.styles'

describe('ModalConfirmacao.styles', () => {
  test('BotaoCancelar DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.BotaoCancelar aoClicar={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  test('BotaoConfirmar DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.BotaoConfirmar aoClicar={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  test('ContainerBotoes DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.ContainerBotoes />)
    expect(container).toMatchSnapshot()
  })

  test('IconeCancelar DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.IconeCancelar />)
    expect(container).toMatchSnapshot()
  })

  test('ModalContent DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.ModalContent />)
    expect(container).toMatchSnapshot()
  })

  test('ModalOverlay DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.ModalOverlay />)
    expect(container).toMatchSnapshot()
  })

  test('TextoMensagem DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.TextoMensagem />)
    expect(container).toMatchSnapshot()
  })

  test('TextoTitulo DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.TextoTitulo />)
    expect(container).toMatchSnapshot()
  })
})
