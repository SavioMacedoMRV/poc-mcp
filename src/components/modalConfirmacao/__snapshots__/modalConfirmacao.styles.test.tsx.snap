// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalConfirmacao.styles BotaoCancelar DEVE ser igual ao snapshot 1`] = `
.c2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  font-family: AvertaStd-Semibold;
  color: #FFFFFF;
  text-align: start;
  cursor: pointer;
  text-transform: none;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: 144px;
  height: 48px;
  cursor: pointer;
  opacity: 1;
  background: #5D615F;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

.c1 {
  border-radius: 16px;
}

<div>
  <button
    altura="48"
    class="c0 c1"
    cor="#5D615F"
    largura="144"
  >
    <span
      alturalinha="24"
      class="c2 "
      cor="#FFFFFF"
      cursor="pointer"
      data-testid="texto-botao"
      estilo="semibold"
      tamanho="16"
    >
      Cancelar
    </span>
  </button>
</div>
`;

exports[`ModalConfirmacao.styles BotaoConfirmar DEVE ser igual ao snapshot 1`] = `
.c2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  font-family: AvertaStd-Semibold;
  color: #E04C36;
  text-align: start;
  cursor: pointer;
  text-transform: none;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: 144px;
  height: 48px;
  cursor: pointer;
  opacity: 1;
  background: #F7F5F2;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

.c1 {
  border-radius: 16px;
}

<div>
  <button
    altura="48"
    class="c0 c1"
    cor="#F7F5F2"
    largura="144"
  >
    <span
      alturalinha="24"
      class="c2 "
      cor="#E04C36"
      cursor="pointer"
      data-testid="texto-botao"
      estilo="semibold"
      tamanho="16"
    >
      Confirmar
    </span>
  </button>
</div>
`;

exports[`ModalConfirmacao.styles ContainerBotoes DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  width: 100%;
  justify-content: space-between;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`ModalConfirmacao.styles IconeCancelar DEVE ser igual ao snapshot 1`] = `
<div>
  <svg
    class=""
    data-file-name="SvgICancelar"
    fill="#434645"
    height="48"
    style="min-width: 48px; min-height: 48px; max-width: 48px; max-height: 48px; transform: rotate(0deg);"
    width="48"
  />
</div>
`;

exports[`ModalConfirmacao.styles ModalContent DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  gap: 32px;
  padding: 32px;
  width: 350px;
  border-radius: 32px;
  background-color: #FFFFFF;
}

@media (max-width: 768px) {
  .c0 {
    width: 300px;
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`ModalConfirmacao.styles ModalOverlay DEVE ser igual ao snapshot 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1;
  width: 100%;
  height: 100%;
  background-color: rgba(93, 97, 95, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`ModalConfirmacao.styles TextoMensagem DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-weight: 400;
  font-size: 14px;
  line-height: normal;
  font-family: AvertaStd-Regular;
  color: #5D615F;
  text-align: center;
  cursor: text;
  text-transform: none;
}

<div>
  <span
    alinhamento="center"
    class="c0 "
    cor="#5D615F"
    tamanho="14"
  />
</div>
`;

exports[`ModalConfirmacao.styles TextoTitulo DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-weight: 400;
  font-size: 18px;
  line-height: normal;
  font-family: AvertaStd-Semibold;
  color: #434645;
  text-align: start;
  cursor: text;
  text-transform: none;
}

<div>
  <span
    class="c0 "
    cor="#434645"
    estilo="semibold"
    tamanho="18"
  />
</div>
`;
